Sources/PT.PM.Common/Nodes/UstNode.cs

```diff
@@ -65,33 +65,7 @@ public virtual int CompareTo(UstNode other)
                 return nodeTypeCompareResult;
             }
 
-            var childrenCountCompareResult = Children.Length - other.Children.Length;
-            if (childrenCountCompareResult != 0)
-            {
-                return childrenCountCompareResult;
-            }
-
-            for (int i = 0; i < Children.Length; i++)
-            {
-                var child = Children[i];
-                if (child == null)
-                {
-                    if (other.Children[i] != null)
-                    {
-                        return -(int)NodeType;
-                    }
-                }
-                else if (child.NodeType != NodeType.FileNode)
-                {
-                    var childCompareResult = child.CompareTo(other.Children[i]);
-                    if (childCompareResult != 0)
-                    {
-                        return childCompareResult;
-                    }
-                }
-            }
-
-            return 0;
+            return UstNodeHelper.CompareCollections(Children, other.Children);
```

Comment:
Maybe transform this method to extension method for shortening code: `Children.Compare(other.Children)`